test/e2e/storage/testsuites/provisioning.go

```diff
@@ -348,21 +379,18 @@ func (t StorageClassTest) TestDynamicProvisioning() *v1.PersistentVolume {
 		}
 	}()
 
-	if class == nil {
-		// StorageClass is nil, so the default one will be used
-		scName, err := e2epv.GetDefaultStorageClassName(client)
-		framework.ExpectNoError(err)
-		defaultSC, err := client.StorageV1().StorageClasses().Get(context.TODO(), scName, metav1.GetOptions{})
+	// ensure that the claim refers to the provisioned StorageClass
+	framework.ExpectEqual(*claim.Spec.StorageClassName, class.Name)
+
+	// If it's the default StorageClass and if late binding is configured, create and delete a pod to provision the volume
+	defaultStorageClassName, err := e2epv.GetDefaultStorageClassName(client)
+	if err != nil && defaultStorageClassName == class.Name && *class.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
```

Comment:
If the input storageclass has WaitForFirstConsumer, then we should do this regardless.

Reply:
I had it like:

```go
	if *class.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
```

But a test was failing, after comparing master I saw that this was only done for the default StorageClass

Reply:
So I pushed a new change with the line above and CI is now failing: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/99346/pull-kubernetes-e2e-gce-storage-snapshot/1365397576798441472/, what should I do in this case? should I use the same code in master (the one that has `if err != nil && defaultStorageClassName == class.Name`) or should I attempt to find the root cause of this bug?

Reply:
I think it would be best to debug why the test is failing.

Reply:
Ack :)